home-service-full-stack/frontend/src/components/booking/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { createBooking, fetchUserBookings } from "./api";
+import { UserContext } from "../context/UserContext";
+import { useContext } from "react";
+import { BookingStatus, BookingWithBusiness } from "./types";
+
+export const BOOKING_KEY = "BOOKING";
+
+export const useUserBookings = (status: BookingStatus) => {
+  const { user } = useContext(UserContext);
+  const userEmail = user!.email;
+
+  return useQuery({
+    queryKey: [BOOKING_KEY, userEmail, status],
+    queryFn: async () => {
+      const bookings: BookingWithBusiness[] = await fetchUserBookings(userEmail, status);
```

Comment:
jeigu nurodyta teisingai timas `fetchUserBookings` `Promise<BookingWithBusinees[]>` tuomet nereikės niekur nurodytinėti papildomai

Reply:
Ištryniau visur, kur nurodytas šis promisas, apart fetchUserBookings